### PR TITLE
Move xhr and reportEvent functions into utils.js.

### PR DESCRIPTION
**Pull request by jimmc**
This moves a couple more functions into utils.js as a cleanup.
It also makes settings.js a module. Unfortunately, I couldn't figure out how to late-load settings.js as a module. If you happen to know how to do that, please let me know.

### CODE REVIEW

File: sources/web/datalab/static/settings.html [hunk 1]
```diff
@@ -13,6 +13,6 @@
       <label><input type="radio" id="darkThemeRadioOption" checked="">Dark Theme</label>
     </div>
   </div>
-  <script>require('static/settings').init();</script>
+  <script>require(['/static/settings.js'],settings => settings.init())</script>
```
Comment (yebrahim) [inline sources/web/datalab/static/settings.html]: nit: space before `settings` function

File: sources/web/datalab/static/settings.js [hunk 2]
```diff
@@ -1,32 +1,42 @@
...
-}
+define(['util'], (util) => {
+  function getSettingKeyAddress(setting) {
+    return window.location.protocol + "//" + window.location.host + "/_settings?key=" + setting;
+  }
 
-// Prepare the theme selector radio boxes
-lightThemeRadioOption = document.getElementById("lightThemeRadioOption")
-darkThemeRadioOption = document.getElementById("darkThemeRadioOption")
+  function init() {
```
Comment (yebrahim) [inline sources/web/datalab/static/settings.js]: probably forgot to remove?

File: sources/web/datalab/static/settings.html [hunk 3]
```diff
@@ -13,6 +13,6 @@
       <label><input type="radio" id="darkThemeRadioOption" checked="">Dark Theme</label>
     </div>
   </div>
-  <script src="/static/settings.js" async="true" defer="true"></script>
+  <script>require(['/static/settings.js'],settings => settings.init())</script>
```
Comment (yebrahim) [inline sources/web/datalab/static/settings.html]: nit: space before `settings` function

File: sources/web/datalab/static/settings.js (hunk 2 above)
Comment (yebrahim) [inline sources/web/datalab/static/settings.js]: If we keep these console logs around, we should probably add a debug querystring or setting somewhere. For now, I think we should take these out

Comment (jimmc) [inline sources/web/datalab/static/settings.js]: Are you referring to the console.log line? It has been removed.

Comment (yebrahim) [inline sources/web/datalab/static/settings.js]: My bad, didn't see the update.

**Pull request merged**